File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

//Calculates how many of each coin is needed
int greedy(int, int);

int main(void)
{
// Prompts the user for change owed 
    float change;
    int check = 0;
    do
    {
        change = get_float("Change owed: ");
        if(change < 0)
        {
            printf("Invalid input detected\n");
        }
        else
        {
            check = 1;
        }
        
    }
    while(check == 0);
// Convert money to nearest penny
    int cents = round(change*100);
    int coins = 0;
    coins += greedy(25, cents);
    cents = cents - (greedy(25, cents)*25);
    coins += greedy(10, cents);
    cents = cents - (greedy(10, cents)*10);
    coins += greedy(5, cents);
    cents = cents - (greedy(5, cents)*5);
    coins += greedy(1, cents);
    printf("%i\n", coins);
}    
    
int greedy(int x, int cents)
{
    int coins = 0;
    int check = 1;
    while(check == 1)
    {
        if(cents >= x)
        {
            cents -= x;
            coins++;
        }
        else
        {
            check = 0;
        }
    }
    return coins;
}

File: pset6/cash/cash.py
from cs50 import get_float


def calc(value, change, total_coins):
    coins = change // value
    change -= (value * coins)
    total_coins += coins
    return total_coins, change


while True:
    change = get_float("Change owed: ")
    if change > 0:
        change *= 100
        break

coins = 0

# Calculate quarters
coins, change = calc(25, change, coins)

# Calculate dimes
coins, change = calc(10, change, coins)

# Calculate nickels
coins, change = calc(5, change, coins)

# Calculate pennies
coins, change = calc(1, change, coins)

coins = int(coins)

print(f"{coins}")